lib/srv/reexec.go

```diff
@@ -1100,12 +1100,35 @@ func ConfigureCommand(ctx *ServerContext, extraFiles ...*os.File) (*exec.Cmd, er
 			ctx.x11rdyw,
 			ctx.errw,
 		},
+		SysProcAttr: new(syscall.SysProcAttr),
 	}
 	// Add extra files if applicable.
 	if len(extraFiles) > 0 {
 		cmd.ExtraFiles = append(cmd.ExtraFiles, extraFiles...)
 	}
 
+	// For remote port forwarding, the child needs to run as the user to
+	// create listeners with the correct permissions.
+	if subCommand == teleport.RemoteForwardSubCommand {
+		localUser, err := user.Lookup(ctx.Identity.Login)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		credential, err := getCmdCredential(localUser)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		if os.Getuid() != int(credential.Uid) || os.Getgid() != int(credential.Gid) {
+			cmd.SysProcAttr.Credential = credential
```

Comment:
Does an empty, non-nil SysProcAttr have a meaning? If not, can't we just set it here instead?
```suggestion
			cmd.SysProcAttr = &syscall.SysProcAttr{Credential: credential}
```

Reply:
+1